src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1581 @@
+
+        enum SortMode : uint16_t
+        {
+            Name,
+            Type,
+            Population,
+            Stations,
+        };
+
+        // 0x00499F53
+        static void prepare_draw(ui::window* self)
+        {
+            common::prepare_draw(self);
+
+            self->widgets[widx::scrollview].right = self->width - 4;
+            self->widgets[widx::scrollview].bottom = self->height - 14;
+
+            // Reposition header buttons
+            self->widgets[widx::sort_town_name].right = std::min(203, self->width - 8);
+
+            self->widgets[widx::sort_town_type].left = std::min(204, self->width - 8);
+            self->widgets[widx::sort_town_type].right = std::min(283, self->width - 8);
+
+            self->widgets[widx::sort_town_population].left = std::min(284, self->width - 8);
+            self->widgets[widx::sort_town_population].right = std::min(353, self->width - 8);
+
+            self->widgets[widx::sort_town_stations].left = std::min(354, self->width - 8);
+            self->widgets[widx::sort_town_stations].right = std::min(423, self->width - 8);
+
+            // Set header button captions
+            self->widgets[widx::sort_town_name].text = self->sort_mode == SortMode::Name ? string_ids::table_header_name_desc : string_ids::table_header_name;
+            self->widgets[widx::sort_town_type].text = self->sort_mode == SortMode::Type ? string_ids::table_header_town_type_desc : string_ids::table_header_town_type;
+            self->widgets[widx::sort_town_population].text = self->sort_mode == SortMode::Population ? string_ids::table_header_population_desc : string_ids::table_header_population;
+            self->widgets[widx::sort_town_stations].text = self->sort_mode == SortMode::Stations ? string_ids::table_header_stations_desc : string_ids::table_header_stations;
+
+            common::repositionTabs(self);
+        }
+
+        // 0x0049A0F8
+        static void draw_scroll(ui::window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+        {
+            auto shade = colour::get_shade(self->colours[1], 3);
+            gfx::clear_single(*dpi, shade);
+
+            uint16_t yPos = 0;
+            for (uint16_t i = 0; i < self->var_83C; i++)
+            {
+                town_id_t townId = self->row_info[i];
+
+                // Skip items outside of view, or irrelevant to the current filter.
+                if (yPos + rowHeight < dpi->y || yPos >= yPos + rowHeight + dpi->height || townId == (uint16_t)-1)
+                {
+                    yPos += rowHeight;
+                    continue;
+                }
+
+                string_id text_colour_id = string_ids::white_stringid2;
+
+                // Highlight selection.
+                if (townId == self->row_hover)
+                {
+                    gfx::draw_rect(dpi, 0, yPos, self->width, rowHeight, 0x2000030);
+                    text_colour_id = string_ids::wcolour2_stringid2;
```

Comment:
This string has been renamed as of #470:
```suggestion
                    text_colour_id = string_ids::wcolour2_stringid;
```